src/api/transactionSubmission/sign.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { Account } from "../../core";
+import { signTransaction } from "../../internal/transactionSubmission";
+import { AccountAuthenticator, AnyRawTransaction } from "../../transactions";
+import { AptosConfig } from "../aptosConfig";
+
+/**
+ * A class to handle all `Sign` transaction operations
+ */
+export class Sign {
+  readonly config: AptosConfig;
+
+  constructor(config: AptosConfig) {
+    this.config = config;
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  transaction(args: { signer: Account; transaction: AnyRawTransaction }): AccountAuthenticator {
+    return signTransaction({
+      ...args,
+    });
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  transactionAsFeePayer(args: { signer: Account; transaction: AnyRawTransaction }): AccountAuthenticator {
+    const { signer, transaction } = args;
+
+    // if "feePayerAddress" is not in "transaction" it means
+    // this is not a fee payer transaction
+    if (!("feePayerAddress" in transaction)) {
```

Comment:
Someday we'll have to create typeguard functions all around to make these nice looking.

Reply:
ohh we can actually have a better way to handle it now - let me change this yack thing